### PR TITLE
[Merged by Bors] - Add demo kafka-druid-earthquake-data

### PR DESCRIPTION
**Pull request by sbernauer**
## Description

Run it from feature branch with
```
cargo r -- --additional-stacks-file stacks/stacks-v1.yaml --additional-demos-file demos/demos-v1.yaml demo install kafka-druid-earthquake-data -k
```

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: demos/demos-v1.yaml [hunk 1]
```diff
@@ -14,3 +14,18 @@ demos:
...
+      - kafka
+      - druid
+      - superset
+      - minio
+      - s3
+      - earthquake
+    manifests:
+      - plainYaml: demos/kafka-druid-earthquake-data/ingest-test-data.yaml # TODO
+      - plainYaml: demos/kafka-druid-earthquake-data/create-druid-ingestion-job.yaml # TODO
+      - plainYaml: demos/kafka-druid-earthquake-data/setup-superset.yaml # TODO
```
Comment (sbernauer) [inline demos/demos-v1.yaml:31]: WIP

File: demos/kafka-druid-earthquake-data/setup-superset.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+kind: Job
+metadata:
+  name: setup-superset
+spec:
+  template:
+    spec:
+      containers:
+        - name: setup-superset
+          image: python:3.10
+          command: ["bash", "-c", "curl -o superset-assets.zip https://raw.githubusercontent.com/stackabletech/stackablectl/demo-kafa-druid-earthquake-data/demos/kafka-druid-earthquake-data/superset-assets.zip && pip install requests==2.22.0 && python /tmp/script/script.py"] # TODO
```
Comment (sbernauer) [inline demos/kafka-druid-earthquake-data/setup-superset.yaml:12]: WIP

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+This guide assumes you already have the demo `kafka-druid-earthquake-data` installed.
+If you don't have it installed please follow the xref:commands/demo.adoc#_install_demo[documentation on how to install a demo].
+To put it simply you have to run `stackablectl demo install kafka-druid-earthquake-data`.
+====
+
+This demo will
+
+* Install the required Stackable operators
+* Spin up the follow data products
+** *Superset*: A modern data exploration and visualization platform. This demo uses it to execute SQL queries on Druid and build dashboards
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:14]:
```suggestion
** *Superset*: A modern data exploration and visualization platform. This demo utilizes Superset to retrieve data from Druid via SQL queries and build dashboards on top of that data
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+** *Kafka*:  A distributed event streaming platform for high-performance data pipelines, streaming analytics and data integration. This demos uses it as a event streaming platform to stream the data in near-realtime
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:15]:
```suggestion
** *Kafka*:  A distributed event streaming platform for high-performance data pipelines, streaming analytics and data integration. This demos uses it as a event streaming platform to stream the data in near real-time
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+** *Druid*: A real-time database to power modern analytics applications. This demo uses it to ingest the near-realtime data from Kafka, store it and enable SQL access to it
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:16]:
```suggestion
** *Druid*: A real-time database to power modern analytics applications. This demo uses it to ingest the near real-time data from Kafka, store it and enable to access the data via SQL
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+ zookeeper  zookeeper    default    zk                 172.18.0.2:32309                                              
+                                                                                                                     
+ minio      minio-druid  default    http               http://172.18.0.4:32306  Third party service                  
+                                    console-http       http://172.18.0.4:31664  Admin user: root, password: rootroot
+----
+
+[NOTE]
+====
+When a product instance has not finished starting yet, the service will have no endpoint.
+Starting all of the product instances might take an considerable amount of time depending on your internet connectivity.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:55]:
```suggestion
Starting all of the product instances might take a considerable amount of time depending on your internet connectivity.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+In case the product is not ready yet a warning might be shown.
+====
+
+== Inspect data in Kafka
+Kafka is used as an event streaming platform to stream the data in near-realtime.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:60]:
```suggestion
Kafka is used as an event streaming platform to stream the data in near real-time.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+All the messages put in and read from Kafka are structured in dedicated queues called topics.
+The testdata will be put in a topic called `earthquakes`.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:62]:
```suggestion
The test data will be put in a topic called `earthquakes`.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+The records get produced (put in) by the testdata-generator and are consumed (read) by Druid afterwards in the same order they where produced.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:63]:
```suggestion
The records get produced (put in) by the test data generator and are consumed (read) by Druid afterwards in the same order they where produced.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+
+As Kafka itself has no webinterface you have to use a Kafka client like https://docs.confluent.io/platform/current/app-development/kafkacat-usage.html[kafkacat].
+To endpoint where Kafka is reachable is given in the `kafka` endpoint from the `kafka` service in your `stackablectl services list` command output (`172.18.0.4:30968` in this case).
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:66]:
```suggestion
In order to connect to Kafka, use the `kafka` endpoint from the `kafka` service in your `stackablectl services list` command output (`172.18.0.4:30968` in this case).
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+$ kafkacat -b 172.18.0.4:30968 -L
+Metadata for all topics (from broker -1: 172.18.0.4:30968/bootstrap):
+ 1 brokers:
+  broker 1001 at 172.18.0.4:30755 (controller)
+ 1 topics:
+  topic "earthquakes" with 1 partitions:
+    partition 0, leader 1001, replicas: 1001, isrs: 1001
+----
+
+You can see the Kafka consists of one broker and the topic `earthquakes` has been created.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:83]:
```suggestion
You can see that Kafka consists of one broker and the topic `earthquakes` has been created.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 12]
```diff
@@ -0,0 +1,292 @@
...
+$ kafkacat -b 172.18.0.4:30968 -C -t earthquakes -o -1 -c 1 \
+    -f 'Topic %t / Partition %p / Offset: %o / Timestamp: %T\n'
+Topic earthquakes / Partition 0 / Offset: 2670281 / Timestamp: 1660571644060
+----
+
+From the output you can see, that the last record was produced at the timestamp `1660571644060` which translates to `Mo 15. Aug 15:54:04 CEST 2022` (using the command `date -d @1660571644`).
+You can also see that it was the record number `2670281` send to this topic, so ~2.6 million million records have been produced so far.
+
+== Druid
+Druid is used to ingest the near-realtime data from Kafka, store it and enable SQL access to it.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:110]:
```suggestion
Druid is used to ingest the near real-time data from Kafka, store it and enable SQL access to it.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 13]
```diff
@@ -0,0 +1,292 @@
...
+The demo has started a ingestion job reading earthquake records from the Kafka topic `earthquakes` and saving it into Druids storage.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:111]:
```suggestion
The demo has started an ingestion job reading earthquake records from the Kafka topic `earthquakes` and saving it into Druids deep storage.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 14]
```diff
@@ -0,0 +1,292 @@
...
+
+image::demo-kafka-druid-earthquake-data/druid_4.png[]
+
+By clicking on the `earthquakes` datasource you can see the segments the datasource consists of.
+In this case the `earthquakes` datasource is partitioned by the year of the earthquake, resulting in 73 segments.
+
+image::demo-kafka-druid-earthquake-data/druid_5.png[]
+
+Druid offers a web-based way of querying the datasources via SQL.
+To achieve this you first have to on `Query` at the top.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:142]:
```suggestion
To achieve this you first have to click on `Query` at the top.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 15]
```diff
@@ -0,0 +1,292 @@
...
+  count(*) as earthquakes
+from earthquakes
+group by 1
+order by 1 desc
+----
+
+image::demo-kafka-druid-earthquake-data/druid_8.png[]
+
+== Superset
+Superset gives the ability to execute SQL queries and build dashboards.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:170]:
```suggestion
Superset provides the ability to execute SQL queries and build dashboards.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 16]
```diff
@@ -0,0 +1,292 @@
...
+=== View charts
+
+The dashboard `Earthquakes` consists of multiple charts.
+To list the charts click on the tab `Charts` at the top.
+
+image::demo-kafka-druid-earthquake-data/superset_5.png[]
+
+// TODO: Fix typo in Chart name
+Click on the Chart `Number of earthquakes my magnitude`.
+On the left side you can modify the chart and click on `Run` to see the effect.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:199]: The typo being capitalization or what?

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 17]
```diff
@@ -0,0 +1,292 @@
...
+
+image::demo-kafka-druid-earthquake-data/superset_6.png[]
+
+=== View distribution on world map
+
+To look at geographical distribution of the earthquakes you have to click on the tab `Charts` at the top again.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:205]:
```suggestion
To look at the geographical distribution of the earthquakes you have to click on the tab `Charts` at the top again.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 18]
```diff
@@ -0,0 +1,292 @@
...
+On the top click on the tab `SQL Lab` -> `SQL Editor`.
+
+image::demo-kafka-druid-earthquake-data/superset_10.png[]
+
+On the left select the database `druid`, the schema `druid` and set `See table schema` to `earthquakes`.
+
+image::demo-kafka-druid-earthquake-data/superset_11.png[]
+
+On the right textbox enter the desired SQL statement.
+If you do not want to make on up you can use the following:
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:234]:
```suggestion
If you do not want to make one up, you can use the following:
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 19]
```diff
@@ -0,0 +1,292 @@
...
+Log in with the credentials username `root`, password `rootroot`.
+
+image::demo-kafka-druid-earthquake-data/minio_2.png[]
+
+Click on the blue button `Browse` on the bucket `druid` and open the folders `data` -> `earthquakes`.
+
+image::demo-kafka-druid-earthquake-data/minio_3.png[]
+
+As you can see druid saved 199MB of data within 73 prefixes (folders).
+On prefix corresponds to on segment which in turn contains all the data of a year.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:263]:
```suggestion
One prefix corresponds to on segment which in turn contains all the data of a year.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 20]
```diff
@@ -0,0 +1,292 @@
...
+If you don't see any years or for files the reason is, that Druid has not saved it's data from memory to the deep storage (in this case S3).
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:264]:
```suggestion
If you don't see any years or files, the reason is that Druid has not saved it's data from memory to the deep storage (in this case S3).
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 21]
```diff
@@ -0,0 +1,292 @@
...
+After waiting a bit the data should have been flushed to S3 and should show up.
+
+image::demo-kafka-druid-earthquake-data/minio_3.png[]
+
+If you open up a prefix for a specific year you can see that Druid has placed a file containing the data of this year there.
+
+== Summary
+The demo streamed 1000 earthquake records/s for a total of ~3 million earthquakes into a Kafka steaming pipeline.
+Druid ingested the data near-realtime into its datasource and enabled SQL access to it.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:273]:
```suggestion
Druid ingested the data near real-time into its data source and enabled SQL access to it.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 22]
```diff
@@ -0,0 +1,292 @@
...
+Superset was used as a web-based frontend to execute SQL statements and build dashboards.
+
+== Where to go from here
+There are multiple paths to go from here.
+The following sections can give you some ideas on what to explore next.
+You can find the description of the earthquake data https://earthquake.usgs.gov/earthquakes/feed/v1.0/csv.php[on the United States Geological Survey website].
+
+=== Execute arbitrary SQL statements
+Within Superset (or the Druid webinterface) you can execute arbitrary SQL statements to explore the taxi data.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:282]:
```suggestion
Within Superset (or the Druid webinterface) you can execute arbitrary SQL statements to explore the earthquake data.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 23]
```diff
@@ -0,0 +1,292 @@
...
+
+=== Create additional dashboards
+You also have the possibility to create additional charts and bundle them together in a Dashboard.
+Have a look at https://superset.apache.org/docs/creating-charts-dashboards/creating-your-first-dashboard#creating-charts-in-explore-view[the Superset documentation] on how to do that.
+
+=== Load additional data
+You can use a kafka client like https://docs.confluent.io/platform/current/app-development/kafkacat-usage.html[kafkacat] to create new topics and  ingest data.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:289]:
```suggestion
You can use a kafka client like https://docs.confluent.io/platform/current/app-development/kafkacat-usage.html[kafkacat] to create new topics and ingest data.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 24]
```diff
@@ -0,0 +1,292 @@
...
+Using the Druid webinterface you can start a ingestion job that consumes the data and stores it in an internal datasource.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:290]:
```suggestion
Using the Druid web interface, you can start an ingestion job that consumes the data and stores it in an internal data source.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 25]
```diff
@@ -0,0 +1,292 @@
...
+There is a great https://druid.apache.org/docs/latest/tutorials/tutorial-kafka.html#loading-data-with-the-data-loader[tutorial] from Druid on how to do this.
+Afterwards the datasource is available for analysis within Druid and Superset the same way the earthquake data is.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:292]:
```suggestion
Afterwards the data source is available to be analyzed within Druid and Superset the same way the earthquake data is.
```

File: docs/modules/ROOT/pages/demos/trino-taxi-data.adoc [hunk 26]
```diff
@@ -54,8 +54,8 @@ In case the product is not ready yet a warning might be shown.
 ====
 
 == Inspect data in S3
-The S3 provided by MinIO is used as a persistent storage to store all the data used
-You can have a look at the testdata within the MinIO Webinterface by opening the given `minio` endpoint `console-http` from the `stackablectl services list` command.
+The S3 provided by MinIO is used as a persistent storage to store all the data used.
+You can have a look at the testdata within the MinIO Webinterface by opening the given `minio` endpoint `console-http` from your `stackablectl services list` command output.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/trino-taxi-data.adoc:58]:
```suggestion
You can have a look at the testdata within the MinIO web interface by opening the given `minio` endpoint `console-http` from your `stackablectl services list` command output.
```

File: docs/modules/ROOT/pages/demos/trino-taxi-data.adoc [hunk 27]
```diff
@@ -11,20 +11,20 @@ This demo will
 
 * Install the required Stackable operators
 * Spin up the follow data products
-** *Superset*: A modern data exploration and visualization platform. This demo uses it to execute SQL queries and build dashboards
+** *Superset*: A modern data exploration and visualization platform. This demo uses it to execute SQL queries on Trino and build dashboards
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/trino-taxi-data.adoc:14]:
```suggestion
** *Superset*: A modern data exploration and visualization platform. This demo utilizes Superset to retrieve data from Trino via SQL queries and build dashboards on top of that data
```

File: docs/modules/ROOT/pages/demos/trino-taxi-data.adoc [hunk 28]
```diff
@@ -97,7 +97,7 @@ Log in with the credentials username `admin`, password `admin`.
 
 image::demo-trino-taxi-data/superset_2.png[]
 
-=== Inspect dashboard
+=== View dashboard
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/trino-taxi-data.adoc:100]:
```suggestion
=== View the dashboard
```

File: docs/modules/ROOT/pages/quickstart.adoc [hunk 29]
```diff
@@ -8,20 +8,21 @@ Please follow the xref:installation.adoc[Installation documentation] to install
...
 They will automatically show up, as `stackablectl` fetches the available list of demos via the internet.
 To list the available demos run the following command:
 
 [source,console]
 ----
-$ stackablectl demos list
+$ stackablectl demo list
 DEMO                                STACKABLE STACK           DESCRIPTION
 trino-taxi-data                     trino-superset-s3         Demo loading 2.5 years of New York taxi data into S3 bucket, creating a Trino table and a Superset dashboard
+kafka-druid-earthquake-data         kafka-druid-superset-s3   Demo ingestion earthquake data into Kafka, streaming it into Druid and creating a Superset dashboard
```
Comment (maltesander) [inline docs/modules/ROOT/pages/quickstart.adoc:20]:
```suggestion
kafka-druid-earthquake-data         kafka-druid-superset-s3   Demo to ingest earthquake data into Kafka, streaming it into Druid and creating a Superset dashboard
```

File: stacks/stacks-v1.yaml [hunk 30]
```diff
@@ -48,9 +49,10 @@ stacks:
               username: superset
               password: superset
               database: superset
-      - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/druid-superset-s3/zookeeper.yaml
-      - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/druid-superset-s3/druid.yaml
-      - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/druid-superset-s3/superset.yaml
+      - plainYaml: stacks/kafka-druid-superset-s3/zookeeper.yaml # TODO
+      - plainYaml: stacks/kafka-druid-superset-s3/kafka.yaml # TODO
+      - plainYaml: stacks/kafka-druid-superset-s3/druid.yaml # TODO
+      - plainYaml: stacks/kafka-druid-superset-s3/superset.yaml # TODO
```
Comment (maltesander) [inline stacks/stacks-v1.yaml:55]: That TODO is for before merging?

File: demos/demos-v1.yaml [hunk 31]
```diff
@@ -14,3 +14,18 @@ demos:
       - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/trino-taxi-data/load-test-data.yaml
       - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/trino-taxi-data/create-table-in-trino.yaml
       - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/trino-taxi-data/setup-superset.yaml
+  kafka-druid-earthquake-data:
+    description:  Demo ingestion earthquake data into Kafka, streaming it into Druid and creating a Superset dashboard
```
Comment (maltesander) [inline demos/demos-v1.yaml:18]:
```suggestion
    description:  Demo to ingest earthquake data into Kafka, streaming it into Druid and creating a Superset dashboard
```

File: docs/modules/ROOT/pages/commands/demo.adoc [hunk 32]
```diff
@@ -11,6 +11,7 @@ To list the available demos, run the following command:
 $ stackablectl demo list
 DEMO                                STACKABLE STACK           DESCRIPTION
 trino-taxi-data                     trino-superset-s3         Demo loading 2.5 years of New York taxi data into S3 bucket, creating a Trino table and a Superset dashboard
+kafka-druid-earthquake-data         kafka-druid-superset-s3   Demo ingestion earthquake data into Kafka, streaming it into Druid and creating a Superset dashboard
```
Comment (maltesander) [inline docs/modules/ROOT/pages/commands/demo.adoc:14]: If you fix this line in demos-v1.yaml do not forget to update here!

**Review by maltesander — CHANGES_REQUESTED** [review]
Demo worked first try. Very nice, the dashboard looks great.
Please have a look sometimes its Job/job, real-time/realtime/near-realtime. Not sure i covered all of that.
Good job anyhow, im fine with merging after the typos/grammar is fixed!

File: demos/demos-v1.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sbernauer) [reply demos/demos-v1.yaml:31]: I have to switch to a URL pointing to the main branch and not to a file on my (our your) disk that only exists within the feature-branch

File: stacks/stacks-v1.yaml (hunk 30 above)
Comment (sbernauer) [reply stacks/stacks-v1.yaml:55]: exactly

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc (hunk 16 above)
Comment (sbernauer) [reply docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc]: typo in chart name has been fixed

File: demos/kafka-druid-earthquake-data/download_earthquake_data.sh [hunk 33]
```diff
@@ -0,0 +1,35 @@
+# This script is not used for the demo
+# Its purpose is to document how to retrieve the used earthquake data
+
+# rm -f 19*.csv 20*.csv earthquakes.csv
```
Comment (maltesander) [inline demos/kafka-druid-earthquake-data/download_earthquake_data.sh:4]: leftover?

File: docs/modules/ROOT/pages/quickstart.adoc [hunk 34]
```diff
@@ -107,5 +108,10 @@ Have a nice day! 👋
 [INFO ] Installed demo trino-taxi-data. Use "stackablectl services list" to list the installed services
 ----
 
+The demos create Kubernetes jobs, that will populate testdata and talk to the installed products to process the data.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/quickstart.adoc:111]:
```suggestion
The demos create Kubernetes jobs, that will populate test data and talk to the installed products to process the data.
```

File: docs/modules/ROOT/pages/quickstart.adoc [hunk 35]
```diff
@@ -107,5 +108,10 @@ Have a nice day! 👋
...
+Until the products are ready, it is completely normal that the pods of these Jobs will fail with an error.
+They will get retried with an exponentially growing backoff time.
+After the products being ready they should turn green and everything should settle down.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/quickstart.adoc:114]:
```suggestion
After the products are ready they should turn green and everything should settle down.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 36]
```diff
@@ -0,0 +1,291 @@
...
+After clicking on the magnification glass to the right side of the `RUNNING` supervisor you can see additional information.
+On the tab `Statistics` on the left you can see the number of processed records as well as the number of errors.
+The statistics show that Druid is currently ingesting 1251 records/s and has ingested 2,1 million records so far.
+All records have been ingested successfully, which is indicated by having no `processWithError`, `thrownAway` or `unparseable`.
+
+image::demo-kafka-druid-earthquake-data/druid_3.png[]
+
+=== Query datasource
+The started ingestion job has automatically created the Druid datasource `earthquakes`.
+You can see the available datasources by clicking on `Datasources` at the top.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:132]: Im pretty sure its data source (when mentioning the druid tab `Datasources` its fine). This is used alot, i would go for a refactor.
```suggestion
=== Query the data source
The started ingestion job has automatically created the Druid data source `earthquakes`.
You can see the available data sources by clicking on `Datasources` at the top.
```

File: docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc [hunk 37]
```diff
@@ -0,0 +1,291 @@
...
+
+=== Execute arbitrary SQL statements
+Within Superset (or the Druid webinterface) you can execute arbitrary SQL statements to explore the earthquake data.
+
+=== Create additional dashboards
+You also have the possibility to create additional charts and bundle them together in a Dashboard.
+Have a look at https://superset.apache.org/docs/creating-charts-dashboards/creating-your-first-dashboard#creating-charts-in-explore-view[the Superset documentation] on how to do that.
+
+=== Load additional data
+You can use a kafka client like https://docs.confluent.io/platform/current/app-development/kafkacat-usage.html[kafkacat] to create new topics and ingest data.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/demos/kafka-druid-earthquake-data.adoc:288]:
```suggestion
You can use a Kafka client like https://docs.confluent.io/platform/current/app-development/kafkacat-usage.html[kafkacat] to create new topics and ingest data.
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Only minor stuff left over

File: demos/kafka-druid-earthquake-data/download_earthquake_data.sh (hunk 33 above)
Comment (sbernauer) [reply demos/kafka-druid-earthquake-data/download_earthquake_data.sh:4]: Nope, it's documented here for development purposes. But we don't want to execute is automatically

Comment (sbernauer) [reply demos/kafka-druid-earthquake-data/download_earthquake_data.sh:5]: Added a comment to clarify

**Review by maltesander — CHANGES_REQUESTED** [review]
One more :D

**Review by maltesander — APPROVED** [review]
Awesome! LGTM.

**Pull request closed without merging**